cmd/oceanbench/set.go

```diff
@@ -368,6 +370,149 @@ func editDevicesHandler(w http.ResponseWriter, r *http.Request) {
 	http.Redirect(w, r, "/set/devices?ma="+ma, http.StatusFound)
 }
 
+// calibrateDevicesHandler handles calibration of controller-type
+// device voltages.
+//
+// Query params:
+//   - ma: MAC address
+//   - vb: 	Battery Voltage
+//   - vnw: Network Voltage
+//   - vp1: Power 1 Voltage
+//   - vp2: Power 2 Voltage
+//   - vp3: Power 3 Voltage
+//   - va: 	Alarm Voltage
+//   - vr: 	Alarm Recovery Voltage
+//
+// NOTE: All voltages are parsed in Volts.
+func calibrateDevicesHandler(w http.ResponseWriter, r *http.Request) {
+	logRequest(r)
+	if r.Method != http.MethodPost {
+		log.Println("calibration request must use POST action")
+		http.Redirect(w, r, "/set/devices?ma="+r.FormValue("ma"), http.StatusSeeOther)
+		return
+	}
+	ctx := context.Background()
+	_, err := getProfile(w, r)
+	if err != nil {
+		if err != gauth.TokenNotFound {
+			log.Printf("authentication error: %v", err)
+		}
+		http.Redirect(w, r, "/", http.StatusUnauthorized)
+		return
+	}
+
+	mac := r.FormValue("ma")
+	vb, err := strconv.ParseFloat(r.FormValue("vb"), 64)
+	if err != nil && vb != 0 {
+		writeDevices(w, r, "unable to parse battery voltage: %v", err)
+		return
+	}
+	vnw, err := strconv.ParseFloat(r.FormValue("vnw"), 64)
+	if err != nil && vnw != 0 {
+		writeDevices(w, r, "unable to parse network voltage: %v", err)
+		return
+	}
+	vp1, err := strconv.ParseFloat(r.FormValue("vp1"), 64)
+	if err != nil && vp1 != 0 {
+		writeDevices(w, r, "unable to parse power 1 voltage: %v", err)
+		return
+	}
+	vp2, err := strconv.ParseFloat(r.FormValue("vp2"), 64)
+	if err != nil && vp2 != 0 {
+		writeDevices(w, r, "unable to parse power 2 voltage: %v", err)
+		return
+	}
+	vp3, err := strconv.ParseFloat(r.FormValue("vp3"), 64)
+	if err != nil && vp3 != 0 {
+		writeDevices(w, r, "unable to parse power 3 voltage: %v", err)
+		return
+	}
+	va, err := strconv.ParseFloat(r.FormValue("va"), 64)
+	if err != nil && va != 0 {
+		writeDevices(w, r, "unable to parse alarm voltage: %v", err)
+		return
+	}
+	vr, err := strconv.ParseFloat(r.FormValue("vr"), 64)
+	if err != nil && vr != 0 {
+		writeDevices(w, r, "unable to parse alarm recovery voltage: %v", err)
+		return
+	}
+
+	device, err := model.GetDevice(ctx, settingsStore, model.MacEncode(mac))
+	if err != nil {
+		writeDevices(w, r, "unable to get device to calibrate (%s): %v", mac, err)
+	}
+
+	// Names of the voltage sensors to calibrate.
+	var voltageSensors = []string{
+		model.NameBatterySensor, model.NameNWVoltage, model.NameP1Voltage,
```

Comment:
This might be a style preference thing, but I reckon it would look slightly nicer if it was listed like this
```
var voltageSensors = []string{
    model.NameBatterySensor,
    model.NameNWVoltage,
....
```